### PR TITLE
feat: improve the chat

### PR DESCRIPTION
**Pull request by workaryangupta**
**Purpose of PR:**

This Pull Request aims to introduce the following enhancements to the chat:

- Create a separate channel for service messages to improve readability by preventing them from interfering with regular chat messages.
- Show user profile images alongside their names in the collaborator list.
- Overlay a green dot on active users' profile image in the collaborator list to visually indicate their current activity status.
- More detailed service messages: they specify what waypoint was moved or deleted, inserted a new waypoint, created a hexagon. For e.g. "{username} moved waypoint 7".

Screenshot 2024-08-18 at 1 17 22 AM

### CODE REVIEW

**Comment by ReimarBauer** [comment]
The changes can be clicked in the list and selected. I think that is currently not needed. 
selection

When possible the output could mention e.g. waypoint 4 moved, new waypoint inserted, waypoint 7 removed.

The active users maybe can written bold in the left list, sorted on top, also avatars maybe become shown.
Also in the Logged in line the avatar could be shown. This Logged in is on many of the views below operations too.

**Comment by ReimarBauer** [comment]
The dot is grey, the picture on top shows it as green one, I would also prefer.

image

**Comment by ReimarBauer** [comment]
looks good :)

example1

**Review by ReimarBauer — COMMENTED** [review]
I do tomm. a further check. Looks pretty good so far. Thx.

File: mslib/msui/mscolab_chat.py [hunk 1]
```diff
@@ -327,19 +329,43 @@ def edit_message(self):
...
-                item = QtWidgets.QListWidgetItem(f'{user["username"]} - {user["access_level"]}',
-                                                 parent=self.collaboratorsList)
+                # Add active status to all the users
+                active_status = "🟢" if user["id"] in active_users else ""
+                display_text = f'{active_status} {user["username"]} - {user["access_level"]}'
+                item = QtWidgets.QListWidgetItem(display_text, parent=self.collaboratorsList)
+
+                # Fetch and set the avatar if it exists
+                if user['avatar_path']:
+                    full_avatar_path = fs.path.join(mscolab_settings.UPLOAD_FOLDER, user['avatar_path'])
```
Comment (ReimarBauer) [inline mslib/msui/mscolab_chat.py]: server request

**Comment by ReimarBauer** [comment]
The dot can have a different size. Can we have it always same size? Example from the seed setup and one user has added a profile image


image

**Comment by ReimarBauer** [comment]
please look also on the table view docking widget "Add Hexagon" This can add a bunch of waypoints, 
image
on deletion in this example it tells only about waypoint 0

maybe use the comment "Hexagon created" in that case for the common insert message.

**Comment by workaryangupta** [comment]
> Can we have it always same size?
Yes I fixed it now

File: mslib/msui/flighttrack.py [hunk 2]
```diff
@@ -352,6 +355,7 @@ def setData(self, index, value, role=QtCore.Qt.EditRole, update=True):
                         waypoint.location = loc[1]
                     # A change of position requires an update of the distances.
                     if update:
+                        self.changeMessageSignal.emit("")
```
Comment (ReimarBauer) [inline mslib/msui/flighttrack.py]: please add a comment why that empty string is important.

**Review by ReimarBauer — COMMENTED** [review]
please add the comment for the empty string

**Review by ReimarBauer — APPROVED** [review]
thx

**Pull request merged**